src/scouter/scanner/ScanningTab.tsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useRef, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import "./QRStyles.css";
+import QrScanner from "qr-scanner";
+import React from "react";
+
+const ScanningTab = () => {
+  const navigate = useNavigate();
+  const scanner = useRef<QrScanner>();
+  const videoEl = useRef<HTMLVideoElement>(null);
+  const qrBoxEl = useRef<HTMLDivElement>(null);
+  const [qrOn, setQrOn] = useState<boolean>(false); // Tracks if the QR scanner is on
+  const [scannedResult, setScannedResult] = useState<string | undefined>("");
+
+  const onScanSuccess = (result: QrScanner.ScanResult) => {
+    console.log(result);
+    const DecodedData = btoa(result.data); 
+    setScannedResult(DecodedData);
+    navigate("/", { state: { result: DecodedData } });
+  };
+
+  const onScanFail = (err: string | Error) => {
+    console.log("QR scan failed:", err);
+  };
+
+  const requestCameraAccess = async () => {
+    console.log("Requesting camera access...");
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia({ video: true });
+      if (videoEl?.current) {
+        console.log("Camera access granted");
+        videoEl.current.srcObject = stream;
+      }
+      initializeScanner();
+    } catch (error) {
+      console.error("Camera access denied:", error);
+      setQrOn(false); // Ensure qrOn is set to false if camera access is denied
+    }
+  };
+
+  const initializeScanner = () => {
+    if (videoEl?.current && !scanner.current) {
+      scanner.current = new QrScanner(videoEl?.current, onScanSuccess, {
+        onDecodeError: onScanFail,
+        preferredCamera: "environment",
+        highlightScanRegion: true,
+        highlightCodeOutline: true,
+        overlay: qrBoxEl?.current || undefined,
+      });
+
+      scanner.current
+        .start()
+        .then(() => setQrOn(true))  // Set qrOn to true when the scanner starts
+        .catch((err) => {
+          console.error("Error starting QR scanner:", err);
+          setQrOn(false);  // Set qrOn to false if the scanner fails to start
+        });
+    }
+  };
+
+  // Cleanup on unmount
+  useEffect(() => {
+    console.log("FIRST");
+    return () => {
+      if (scanner?.current) {
+        console.log("Stopping scanner on unmount...");
+        scanner.current.stop();
+      }
+    };
+  }, []);
+
+  // Alert user if the camera is blocked
+  useEffect(() => {
+    if (!qrOn) {
+      console.log("QR scanner is off or camera is blocked.");
+    }
+  }, [qrOn]);
```

Comment:
doing `console.log()` does not count as alerting the user, consider using `alert()` or another alternative that is visible without opening dev tools